031-tls-encrypt-the-kafka-connect-rest-api.md

```diff
@@ -0,0 +1,161 @@
+                enum:
+                - http
+                - https
+                description: The protocol of the REST listener.
+              certChainAndKey:
+                type: object
+                properties:
+                  certificate:
+                    type: string
+                    description: The name of the certificate file in the Secret.
+                  key:
+                    type: string
+                    description: The name of the private key file in the Secret.
+                  secretName:
+                    type: string
+                    description: The name of the Secret containing the certificate.
+                  required:
+                    - key
+                    - certificate
+                    - secretName
+                description: Certificate for Kafka Connect to use to create a keystore for the rest listener.
+              strimziTruststoreCert:
+                type: object
+                properties:
+                  certificate:
+                    type: string
+                    description: The name of the certificate file in the Secret.
+                  secretName:
+                    type: string
+                    description: The name of the Secret containing the certificate.
+                  required:
+                    - certificate
+                    - secretName
+                description: Certificate for Strimzi to trust when making calls to the Connect rest listener.
+            required:
+            - protocol
+          description: List of additional REST listeners.
+        ...
+```
+
+the above CR would result in the following properties in the generated connect configuration:
+
+```
+listeners: https://:8443,http://:8083
+rest.advertised.listener: https
+rest.advertised.port: 8443
+listeners.https.ssl.client.auth: none
+listeners.https.ssl.keystore.location: /tmp/kafka/kafka-connect-rest.keystore.p12
+listeners.https.ssl.keystore.password: ***generated password***
+listeners.https.ssl.keystore.type: PKCS12
+```
+
+### Certificates
+
+If the user decides to enable an encrypted HTTPS listener, they must have the `certChainAndKey` 
+property present in the `KafkaConnect` CR. The property must point to an existing secret that contains the 
+certificate and private key that they want Kafka Connect to use. The certificate must contain the correct 
+Subject Alternative Names (SANs) similarly to when a user provides their own certificates for listeners:
+https://strimzi.io/docs/operators/latest/using.html#ref-alternative-subjects-certs-for-listeners-str
+
+If the user has enabled an encrypted HTTPS listener and the certificate they referenced in the `certChainAndKey` field 
+is not signed by a well known CA, they must also configure the `strimziTruststoreCert`. This property must point to an 
+existing secret that contains a certificate that Strimzi can use to generate a truststore. The resulting truststore will 
```

Comment:
Can't we use `trustedCertificates` instead of `strimziTruststoreCert`, I am not sure how much it makes sense having "strimzi" in the name and how much it could be confusing for users.
The `trustedCertificates` name is the same we use in other place of our CRs like bridge connecting to Kafka (so trusting the Strimzi CA for example) or even the same Kafka Connect connecting to Kafka (see `tls.trustedCertificates` in the spec).
I would also specified that the contained certificate was used to sign the cert provided in the `certChainAndKey`, not just saying "that Strimzi can use to generate a truststore" which sounds to be too much general.